app/services/person_constructor.rb

```diff
@@ -0,0 +1,82 @@
+class PersonConstructor
+  KEY_NAME_MAPPINGS = {
+    address: :address_1,
+    zip: :zip_code,
+  }
+  OLD_REMOTE_PARAMS = [
+    remote_fields: PersonWithRemoteFields::REMOTE_PARAMS
+  ]
+
+  def self.permitted_params
+    Person::PERMITTED_PARAMS +
+      PersonWithRemoteFields::REMOTE_PARAMS +
+      Location::PERMITTED_PARAMS +
+      KEY_NAME_MAPPINGS.keys +
+      OLD_REMOTE_PARAMS +
+      [:uuid] # this should come from PersonFinder
+  end
+
+  def initialize(params)
+    @params = params.deep_symbolize_keys
+    normalize_params
+  end
+
+  def build
+    person = find_or_initialize_person_with_remote_fields
+    person.assign_attributes(person_params)
+    assign_location_attributes(person)
+    person
+  end
+
+  private
+
+  attr_reader :params
+
+  def find_or_initialize_person_with_remote_fields
+    if person = PersonFinder.new(params).find
+      person.becomes(PersonWithRemoteFields)
+    else
+      PersonWithRemoteFields.new
+    end
+  end
+
+  def assign_location_attributes(person)
+    if location_params.any?
+      new_location = LocationComparable.new(location_params)
+      person.location.becomes(LocationComparable).merge(new_location)
+      person.location.fill_in_missing_attributes
+    end
+  end
+
+  def normalize_params
+    flatten_remote_fields
+    rename_keys
+    strip_whitespace_from_values
+  end
+
+  def flatten_remote_fields
+    params.merge!(params.delete(:remote_fields) || {})
+  end
+
+  def strip_whitespace_from_values
+    params.merge!(params){ |k, v1| v1.try(:strip) || v1 }
+  end
+
+  def rename_keys
```

Comment:
I might just call this method `map_local_keys` or `map_remote_keys` in an attempt  more specificly describe the purpose of this mutation.